mypy/messages.py

```diff
@@ -30,7 +30,7 @@
 INVALID_EXCEPTION = 'Exception must be derived from BaseException'
 INVALID_EXCEPTION_TYPE = 'Exception type must be derived from BaseException'
 INVALID_RETURN_TYPE_FOR_YIELD = \
-    'Iterator function return type expected for "yield"'
+    'Generator or supertype function return type expected for "yield"'
```

Comment:
This sentence sounds a bit odd. Maybe we can throw it around, e.g.
```
The return type of a generator function should be "Generator" or a supertype thereof.
```
(The mention of "yield" in the original message is kind of a distraction -- the problem is not with the yield statement/expression but with the function's type declaration. Extra points if you can find the line number of the 'def' line.)